ion/writer.go

```diff
@@ -148,40 +146,22 @@ type writer struct {
 	ctx ctxstack
 	err error
 
-	fieldName   *string
+	fieldName   *SymbolToken
 	annotations []SymbolToken
 }
 
-// FieldName sets the field name for the next value written.
+// FieldName sets the field name symbol for the next value written.
 // It may only be called while writing a struct.
-func (w *writer) FieldName(val string) error {
+func (w *writer) FieldName(val SymbolToken) error {
 	if w.err != nil {
 		return w.err
 	}
-	if !w.IsInStruct() {
-		w.err = &UsageError{"Writer.FieldName", "called when not writing a struct"}
-		return w.err
-	}
-
-	w.fieldName = &val
-	return nil
-}
-
-// FieldNameSymbol sets the field name symbol for the next value written.
-// It may only be called while writing a struct.
-func (w *writer) FieldNameSymbol(val SymbolToken) error {
 	if !w.IsInStruct() {
 		w.err = &UsageError{"Writer.FieldNameSymbol", "called when not writing a struct"}
```

Comment:
"Writer.FieldName" instead of "Writer.FieldNameSymbol"?

Reply:
Addressed.